src/SqlPersistence/Config/SqlPersistenceConfig_SqlDialect_MsSqlServer.cs

```diff
@@ -11,8 +11,16 @@ public static void Schema(this SqlDialectSettings<SqlDialect.MsSqlServer> dialec
             Guard.AgainstNull(nameof(dialectSettings), dialectSettings);
             Guard.AgainstNullAndEmpty(nameof(schema), schema);
             Guard.AgainstSqlDelimiters(nameof(schema), schema);
-            dialectSettings.Settings.Schema = schema;
+            dialectSettings.TypedDialect.Schema = schema;
         }
 
+        /// <summary>
+        /// Instructs the persistence to not use the connection established by the SQL Server transport.
```

Comment:
@DavidBoike that used to be an option with NHibernate but since SQL persistence uses `Func<DbConnection>` it would mean that for each message we need to create the persistence connection using the factory and check if connection string's match. 

Alternatively we can *assume* that the connection factory always returns connection with the same connection string and check only once but I am not sure if we can make such an assumption.